src/main/java/edu/wpi/grip/ui/PaletteView.java

```diff
@@ -37,9 +49,46 @@ public PaletteView(EventBus eventBus) {
             throw new RuntimeException(e);
         }
 
+
+        try {
+            setupClearButtonField(operationSearch);
+        } catch (Exception e) {
+            throw new IllegalStateException("Problem calling the setupClearButtonField method on TextFields", e);
+        }
+
+        final InvalidationListener filterOperations = observable ->
+                this.operations.getChildren().forEach(node -> {
+                    final Operation operation = ((OperationView) node).getOperation();
+                    node.setVisible(searchAlgorithmMatches(this.operationSearch.getText(), operation.getName()));
+                });
+
+        // Update the visibility of each node when either the list of nodes or the filter text changes
+        this.operations.getChildren().addListener(filterOperations);
+        this.operationSearch.textProperty().addListener(filterOperations);
+
         this.eventBus.register(this);
     }
 
+
+    // XXX: This is the only way to do this unfortunately.
+    // This is where this came from: https://bitbucket.org/controlsfx/controlsfx/issues/330/making-textfieldssetupclearbuttonfield
+    private void setupClearButtonField(CustomTextField customTextField) throws Exception {
+        Method m = TextFields.class.getDeclaredMethod("setupClearButtonField", TextField.class, ObjectProperty.class);
```

Comment:
I actually didn't even notice the clear button at first.  Would it be better to do something like this?  It's infinitely more cleaner than using reflection, and I like it better because it shows an actual button that says "clear".

Palette.fxml

        <CustomTextField fx:id="search" promptText="Search">
            <right>
                <Button fx:id="clearSearch" text="clear"/>
            </right>
        </CustomTextField>

PaletteView.java

    @FXML
    private TextInputControl search;

    @FXML
    private Button clearSearch;

...

        this.clearSearch.setOnMouseClicked(event-> this.search.clear());
        this.clearSearch.visibleProperty().bind(this.search.textProperty().isEmpty().not());